tikv/region.go

```diff
@@ -455,29 +455,7 @@ func (rm *RegionManager) getRegionFromCtx(ctx *kvrpcpb.Context) (*regionCtx, *er
 		// Wait for the parent region reference decrease to zero, so the memLocks would be clean.
 		// We are not going to face any
 		parent.refCount.Wait()
-		if atomic.CompareAndSwapPointer(&ptr, unsafe.Pointer(parent), nil) {
-			// Inherent the txnKeysMap from parent.
-			isLeft := bytes.Equal(parent.startKey, ri.startKey)
-			ri.txnKeysMu.Lock()
-			for startTS, parentKeys := range parent.txnKeysMap {
-				var newKeys [][]byte
-				for _, parentKey := range parentKeys {
-					if isLeft {
-						if bytes.Compare(parentKey, ri.endKey) < 0 {
-							newKeys = append(newKeys, parentKey)
-						}
-					} else {
-						if bytes.Compare(parentKey, ri.startKey) >= 0 {
-							newKeys = append(newKeys, parentKey)
-						}
-					}
-				}
-				if len(newKeys) > 0 {
-					ri.txnKeysMap[startTS] = newKeys
-				}
-			}
-			ri.txnKeysMu.Unlock()
-		}
+		atomic.StorePointer(&ptr, nil)
```

Comment:
Could add a todo here? We are moving to next stage, we still need to handle lock split.